mlflow/utils/environment.py

```diff
@@ -194,7 +195,13 @@ def infer_pip_requirements(model_uri, flavor, fallback=None):
         return _infer_requirements(model_uri, flavor)
     except Exception:
         if fallback is not None:
-            _logger.exception(_INFER_PIP_REQUIREMENTS_FALLBACK_MESSAGE, model_uri, flavor)
+            if _logger.level <= logging.DEBUG:
+                _logger.debug(_INFER_PIP_REQUIREMENTS_FALLBACK_MESSAGE,
+                              model_uri, flavor, exc_info=True)
+            else:
+                _logger.warning(_INFER_PIP_REQUIREMENTS_FALLBACK_MESSAGE +
+                                ' Set log level to be DEBUG to get more information.',
+                                model_uri, flavor)
```

Comment:
```suggestion
            _logger.warning(_INFER_PIP_REQUIREMENTS_FALLBACK_MESSAGE +
                            ' Set log level to be DEBUG to get more information.',
                                model_uri, flavor, fallback)
            _logger.debug("", exc_info=True)
```

Reply:
`_logger.debug("", exc_info=True)` prints out nothing unless the logging level is set to DEBUG.